VRCFaceTracking/Params/Expressions/UnifiedExpressions.cs

```diff
@@ -0,0 +1,201 @@
+﻿namespace VRCFaceTracking.Params
+{
+    // TODO: Create expressions map to relate legacy shapes to new ones
+
+    /// <summary> 
+    /// Represents the type of Shape data being sent by UnifiedExpressionData, in the form of enumerated shapes.
+    /// </summary>
+    public enum UnifiedExpressions
+    {
+        #region Eye Gaze Expressions
+
+        // These are currently unused for expressions and used in the UnifiedEye structure.
+        // EyeLookOutRight,
+        // EyeLookInRight,
+        // EyeLookUpRight,
+        // EyeLookDownRight,
+        // EyeLookOutLeft,
+        // EyeLookInLeft,
+        // EyeLookUpLeft,
+        // EyeLookDownLeft,
+
+        #endregion
+
+        #region Eye Expressions
+
+        // 'Biometrically' accurate data that is included with UnifiedEye
+        //EyeClosedRight, // Closes the right eyelid
+        //EyeClosedLeft, // Closes the left eyelid
+        //EyeDilationRight, // Dilates the right eye's pupil
+        //EyeDilationLeft, // Dilates the left eye's pupil
+        //EyeConstrictRight, // Constricts the right eye's pupil
+        //EyeConstrictLeft, // Constricts the left eye's pupil
+
```

Comment:
Add Squeeze. Squint and squeeze are not the same

Reply:
Could you elaborate further, I'm not entirely sure what you consider them different. Here is what I've considered when using this parameter for this particular v1 expression:

Unified Squint = Quest Pro Lid Tightener = AR52 Eye Squint = (SRan Eye Squeeze + SRan Eyes Closed)

SRanipal EyeSqueeze from what I can tell from using and from visual references is a combination of squinting the lower eyelid muscles (Ala Quest Pro Lid Tightener) and closing the eyes (Ala Quest Pro EyeBlink). The only issue is that SRanipal EyeSqueeze only activates when the eyes are closed and acts as it's own threshold beyond closed eyes. I think even from our own SRanipal documentation that this would be the case as well https://github.com/benaclejames/VRCFaceTracking/wiki/Blend-Shapes-Setup#sranipal-eye-shapes

When messing around with the High Fidelity Quest blendshapes reference it does seem to indicate that when Lid tightener and Eyes closed are active at the same time that there would be a corrective shape that makes them visually identical or of similar function to SRanipal EyeSqueeze.

https://user-images.githubusercontent.com/74634856/214394165-bbd35bcf-e672-4026-a535-d1c82d50190c.mp4

I think the only problem is that this SRanipal Squeeze parameter kind of is locked in to behave properly for SRanipal and doesn't take into account having Eye Blink and Squint be mutually exclusive (which is true for Quest Pro and every AR52 implementation). The SRanipal combined eye parameters would have to be fundamentally changed in order for it to behave as expected with the new standard.

SRanipal Squeeze does track differently but should fundamentally control to the same or very similar shapes to our standard. I am not entirely sure if adding an extra parameter that only SRanipal could solely take advantage of would be necessary if it already has such a huge overlap with the existing parameters. I think we should be building Unified to be able to control SRanipal parameters in a high fidelity way rather than allowing SRanipal to just pass-through directly.

Let me know if you think I am wrong about these assumptions.

Reply:
I also think that the SRanipal emulator could and should be able to replicate SRanipal Eye Squeeze's behavior, I just didn't have a tracking interface that let me play around with the emulator beyond the SRanipal interface until I made the MeowFace interface for testing it.

Reply:
Okay after playing around with the Unified data a bit I believe I have a solution that can solve these two issues, which will allow any tracking interface to still work really well even in the emulated combined eyelids.

I will treat the Squeeze section and Blink + Widen sections of the parameters in their own distinctive ways: 

Blink + Widen will still have Squint weighted into the openness (as it does affect the SRanipal eye shape slightly on all interfaces, it's just that SRanipal doesn't squint with eyeSqueeze) but to a much lesser extent than it currently is enforced in. When the eyes are closed then we can enforce Squint back into it's full value to enact on SRanipal Squeeze. To achieve this I will be putting Squint through an exponential curve in-which the curve will be influenced by Openness and EyeWide. 

I will play around with this approach and see if it achieves the results that I am after.

Reply:
Hmm not sure how it gets tracked on IPhone ARKit but from testing MeowFace I notice that Eye Squint gets canceled out once my eyes blink, which seems to be of a lesser fidelity than the Quest Pro's Lid Tightener and a complete inverse of SRanipal Eye Squeeze. Seems to be a limitation of this tracking interface rather than the standard. Still believe we should consolidate all of these into a single Squint parameter in the new standard.

Reply:
This equation does what I want:
EyeLidExpandedSqueeze (v1): (Openness * 0.8 + Widen * 0.2) - (1 - ((Openness^0.25)) * Squint)

Basically Squint won't affect the eye shape too much until the eyes start closing. Once the eyes are fully closed Squint will have full range to enact Squeeze. Should also fix the Quest Pro squinting problem for this specific parameter.

Reply:
With the Quest Pro demo avatar be careful to avoid the corrective shapes with making the basis for the code. Here is my experience how the Quest Pro tracks squint and eye close:
- Eye close for Quest Pro is Non-linear, it is confidence of Blinking not closing so when blinking it is strong value of 1
- Closing the eyes slow is very insensitive when closing will only get ~0.5 raw value
- Squint triggers strongly when closing the eyes slowly and is mostly saturated when you completely close the eyes. There is no more tracking squint once the eye is closed and you squeeze your eye.
- I have been using squint to help close the eye for the Quest Pro module by add the value to eye closed. You can see previous comment to bring the code back.

Reply:
> With the Quest Pro demo avatar be careful to avoid the corrective shapes with making the basis for the code

I think in this case and many other cases the overlapping shape corrections are an indication of their intention to potentially have the output be able to do those shapes in the future. Basically we should have our standard be capable of that level of fidelity but not really enforce tracking standards to track in that way.

> * Squint triggers strongly when closing the eyes slowly and is mostly saturated when you completely close the eyes. There is no more tracking squint once the eye is closed and you squeeze your eye.

This seems to be a similar behavior that AR52 (specifically in this case MeowFace) has as well, where specifically lifting your lower eyelid and blinking can affect it. Avatars should be built to accept any sort of mixing of these parameters, and in the case of the Quest Pro demos they seem to be able to accept that kind of mixing of data. Squint in Unified will treat AR52/Meta FACS Squint appropriately as well as SRan Eye Squeeze, as this can be considered a tracking limitation of the interfaces in question and should still interface into our current standard properly with intended tracking.

> * I have been using squint to help close the eye for the Quest Pro module by add the value to eye closed. You can see previous comment to bring the code back.

I don't have a Quest Pro to test with but I do remember this was something that was done in the Quest Pro demos when looking at the source scripts. I have reverted this behavior.